lib/graph3d/DataGroup.js

```diff
@@ -0,0 +1,240 @@
+      this.yBarWidth = graph3d.defaultYBarWidth;
+    }
+    else {
+      this.yBarWidth = this.getSmallestDifference(data, this.colY) || 1;
+    }
+  }
+
+  // calculate minimums and maximums
+  var NUMSTEPS = 5;
+
+  var xRange = this.getColumnRange(data, this.colX);
+  if (withBars) {
+    xRange.expand(this.xBarWidth / 2);
+  }
+  this._setRangeDefaults(xRange, graph3d.defaultXMin, graph3d.defaultXMax);
+  this.xRange = xRange;
+  this.xStep = (this.defaultXStep !== undefined) ? graph3d.defaultXStep : xRange.range()/NUMSTEPS;
+
+  var yRange = this.getColumnRange(data, this.colY);
+  if (withBars) {
+    yRange.expand(this.yBarWidth / 2);
+  }
+  this._setRangeDefaults(yRange, graph3d.defaultYMin, graph3d.defaultYMax);
+  this.yRange = yRange;
+  this.yStep = (graph3d.defaultYStep !== undefined) ? graph3d.defaultYStep : yRange.range()/NUMSTEPS;
+
+  var zRange = this.getColumnRange(data, this.colZ);
+  this._setRangeDefaults(zRange, graph3d.defaultZMin, graph3d.defaultZMax);
+  this.zRange = zRange;
+  this.zStep = (graph3d.defaultZStep !== undefined) ? graph3d.defaultZStep : zRange.range()/NUMSTEPS;
+
+  if (data[0].hasOwnProperty('style')) {
+    this.colValue = 'style';
+    var valueRange = this.getColumnRange(data, this.colValue);
+    this._setRangeDefaults(valueRange, graph3d.defaultValueMin, graph3d.defaultValueMax);
+    this.valueRange = valueRange;
+  }
+};
+
+
+DataGroup.prototype.getDistinctValues = function(column) {
+  var data           = this.dataTable;
+  var distinctValues = [];
+
+  for (var i = 0; i < data.length; i++) {
+    if (distinctValues.indexOf(data[i][column]) == -1) {
+      distinctValues.push(data[i][column]);
+    }
+  }
+
+  return distinctValues.sort(function(a,b) { return a - b; });
+};
+
+
+/**
+ * Determine the smallest difference between the values for given
+ * column in the passed data set.
+ *
+ * @returns {Number|null} Smallest difference value or
+ *                        null, if it can't be determined.
+ */
+DataGroup.prototype.getSmallestDifference = function(data, column) {
+  var values = this.getDistinctValues(data, column);
+  var diffs  = [];
+
+  // Get all the distinct diffs
+  // Array values is assumed to be sorted here
+  var smallest_diff = null;
+
+  for (var i = 1; i < values.length; i++) {
+    var diff = values[i] - values[i - 1];
+
+    if (smallest_diff == null || smallest_diff > diff ) {
+      smallest_diff = diff;
+    }
+  }
+
+  return smallest_diff;
+}
+
+
+/**
+ * Get the absolute min/max values for the passed data column.
+ *
+ * @returns {Range} A Range instance with min/max members properly set.
+ */
+DataGroup.prototype.getColumnRange = function(data, column) {
+  var range  = new Range();
+
+  // Adjust the range so that it covers all values in the passed data elements.
+  for (var i = 0; i < data.length; i++) {
+    var item = data[i][column];
+    range.adjust(item);
+  }
+
+  return range;
+};
+
+
+/**
+ * NOTE: Not used yet but it will be.
```

Comment:
Is this comment old? Is it used yet? Do we still need this comment? I suggest to remove the comment and add `@return` since the function is public to the outside.